src/base/cost_functions.h

```diff
@@ -267,6 +267,110 @@ class RelativePoseCostFunction {
   const double y2_;
 };
 
+// Position alignment cost function (position meas. expected to be in world
+// frame).
+class PositionAlignCostFunction {
+ public:
+  explicit PositionAlignCostFunction(
+      const Eigen::Vector3d& pos_xyz,
+      const Eigen::Vector3d& std_xyz = Eigen::Vector3d::Ones())
+      : pos_x_(pos_xyz(0)),
+        pos_y_(pos_xyz(1)),
+        pos_z_(pos_xyz(2)),
+        inv_std_x_(1. / std_xyz(0)),
+        inv_std_y_(1. / std_xyz(1)),
+        inv_std_z_(1. / std_xyz(2)) {}
+
+  static ceres::CostFunction* Create(
+      const Eigen::Vector3d& pos_xyz,
+      const Eigen::Vector3d& std_xyz = Eigen::Vector3d::Ones()) {
+    return (new ceres::AutoDiffCostFunction<PositionAlignCostFunction, 3, 4, 3>(
+        new PositionAlignCostFunction(pos_xyz, std_xyz)));
+  }
+
+  template <typename T>
+  bool operator()(const T* const qvec, const T* const tvec,
+                  T* residuals) const {
+    // Get Pose in world frame (cam -> world transform)
+    Eigen::Matrix<T, 3, 3, Eigen::RowMajor> Rciw;
+    ceres::QuaternionToRotation(qvec, Rciw.data());
+
+    const Eigen::Matrix<T, 3, 1> tciw(tvec[0], tvec[1], tvec[2]);
```

Comment:
Seems like we could instead invert the quaternion and directly rotate the point instead of going through a rotation matrix?